haskellPackages: stackage LTS 24.10 -> LTS 24.12

sternenseemann opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `staging`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `staging` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #429810. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

emilazy committed:
haskell-modules/configuration-hackage2nix: drop mentions of 8.10

sternenseemann committed:
haskellPackages.mkDerivation: enforce unix line endings in cabal files

Automatically use sed(1) to remove DOS line endings from .cabal files.
This is intended to work around Hackage producing revised cabal files
with DOS line endings (https://github.com/haskell/hackage-server/issues/316).
This frequently leads to

- patches failing to apply. Unfortunately there is no way to apply a
  patch with Unix line endings to a file with DOS line endings with
  GNU patch(1). Only the following combinations are supported:

  file  patch  comment
  Unix  Unix
  DOS   DOS    using --binary
  DOS   Unix   Haven't tested it, but I assume it works because
               patch(1) converts the file to Unix line endings
               implicitly (which can be disabled using --binary),
               but never the patch…

- Test suites sometimes involve the .cabal file which can be
  affected by the line endings.

This behavior is enabled by default to prevent cases where (unrevised)
packages regress when they get a revision later. Of course, the behavior
is a little nonsensical for non-Hackage packages, but it is probably
rarely an issue in practice. When DOS line endings need to be preserved,
the user can pass `dontConvertCabalFileToUnix = true`.

Supersedes #279248.
Resolves #206840.

sternenseemann committed:
stack: provide hpack 0.38.1 to match upstream binaries

(cherry picked from commit 4ede35a36fc0bf48bc0ce3fe31c30c64f8427348)

sternenseemann committed:
emanote: remove at 1.4.0.0

The former maintainer and upstream author of this package has been
banned from Nixpkgs f03d404df9c59a2cbb7f6f902247841245c97c3c6ad90.
I have zero interest in keeping this working.

ema happens to work without overrides, so let's leave that enabled until
it breaks.

maralorn committed:
haskell.packages.ghc{94,96,98,912}.haskell-language-server: Fix build

(cherry picked from commit 3bce98bfed6c26282c301505b9e12d8c44767f1d)

sternenseemann committed:
dhall-lsp-server: apply patch for text >= 2.1.2

(cherry picked from commit 7adbd08882e1945cca0f322b8d32509681606526)

sternenseemann committed:
dhall-lsp-server: use lsp from Stackage LTS 24

(cherry picked from commit 1fe73e3aa7956c3178b2ccfd53a788ad26f02b79)

sternenseemann committed:
haskellPackages.{arbtt,binrep,cabal-cargs,dataframe,monad-bayes,patat,regression-simple}: jailbreak

haskellPackages.cabal-cargs: drop override obsoleted by 1.7.0.

(cherry picked from commit d00493bd14a01107c279f251f8fcc264bc6a31a5)

sternenseemann committed:
ihaskell: re-enable ihaskell-diagrams

(cherry picked from commit d74914e95884b2628eacc1fed8042a9c604ef58d)

nixpkgs-ci[bot] committed:
Merge 7b38f24f2fe65231f1f957d7259e929acb260c1f into haskell-updates

sternenseemann committed:
haskellPackages.dunai: drop obsolete workaround

The issue has been resolved upstream as reported by krank.

sternenseemann committed:
haskellPackages.HsSyck: remove obsolete workaround

The issue has been resolved upstream as reported by krank.

sternenseemann committed:
haskellPackages.lvar: drop obsolete override

The issue has been resolved upstream as reported by krank.

sternenseemann committed:
haskellPackages.glualint: drop obsolete override

The issue has been resolved in the last two releases, however glualint
is still transitively broken by
<https://github.com/UU-ComputerScience/uuagc/issues/20>.

sternenseemann committed:
haskellPackages.aeson-schema: drop obsolete override

aeson-schema has been marked broken due to unrelated issues:
https://github.com/ocramz/aeson-schema/issues/26

sternenseemann committed:
haskellPackages.pandoc-crossref: revert patch requiring pandoc>=3.8

- The linked issue has been resolved, but the updated bounds are only
  available on master.
- 0.3.21 contains a patch for an interface deprecation in pandoc 3.8
  which we need to revert in order to build with pandoc 3.7.

(cherry picked from commit f2b98de78a3a2967cf011f10beacef2088daf0f2)

sternenseemann committed:
haskell.compiler: make krank ignore informational issue references

A lot of the links to issues in pkgs/development/compilers/ghc note
historical issues we need to workaround for outdated versions of GHC
or note related discussion. Them being closed is not actionable in many
cases so I have marked them with krank:ignore-line, so they don't show
up in reports generated using krank.

sternenseemann committed:
doc/haskell: make krank ignore informational issue reference

sternenseemann committed:
haskell.compiler: remove references xattr missing flag issues

darwin.xattr used to be a fork of python3.pkgs.xattr.
bafc6ff88df03e765a35efd140e switched darwin.xattr to Apple's new
C implementation of the tool.

When I originally packaged darwin.xattr (283d622397539f196416657d),
I linked tickets on the original tool's issue tracker suggesting to
add support for flags added by Apple in their fork. This has since
happened, so we could possibly switch to python3.pkgs.xattr.

However, I think it's no longer a good idea:

- Bootstrapping-wise a C tool is considerably simpler.
- GHC upstream tests against Apple's xattr(1) implementation in their
  CI. Especially now, since they no longer share a direct ancestry,
  it seems unwise to assume the tools will behave the same.

mpscholten committed:
haskellPackages.ihp-openai: unbreak

mpscholten committed:
haskellPackages.ihp-openai: add maintainer mpscholten

sternenseemann committed:
test.haskell: make krank ignore purely informational issue reference

sternenseemann committed:
haskellPackages.blockio-uring: update issue reference

Upstream added support for liburing 2.11, but since then we have updated
to liburing 2.12.

sternenseemann committed:
haskellPackages: stackage LTS 24.10 -> LTS 24.11

all-cabal-hashes: 2025-09-14T21:34:10Z -> 2025-09-21T18:48:01Z

(generated by maintainers/scripts/haskell/update-package-set.sh)

sternenseemann committed:
haskellPackages.cabal2nix-unstable: use correct output on Darwin

cabal2nix doesn't have a separate bin output, but cabal2nix-unstable
does (now?), so copying over the code doesn't work. Just stop hardcoding
the bin output in both cases.

(cherry picked from commit ec426cc387bba05f0ee92356b06eefb2ceb0b8b2)

nixpkgs-ci[bot] committed:
Merge 6d7dc316d61add37789c272a6d23dc591ef42f66 into haskell-updates

sternenseemann committed:
haskellPackages.math-functions: drop obsolete workaround on darwin

Upstream issue has been resolved.

sternenseemann committed:
haskellPackages.serialport: explain why we skip the test suite

sternenseemann committed:
haskellPackages.GLFW-b: explain why we skip the test suite

sternenseemann committed:
haskell.compiler.ghc9{4,63}: let krank ignore issue ref in patch

This is a historical issue that we no longer need workarounds for with
recent GHC versions.

sternenseemann committed:
nixfmt-classic: note current ticket for bounds issue

sternenseemann committed:
haskellPackages.pfile: note current ticket for bounds issue

sternenseemann committed:
haskellPackages.regex-applicative-text:note it won't receive updates

Closed issue indicates WONTFIX, so we can make krank ignore it.

sternenseemann committed:
haskellPackages.lzma-conduit: make krank ignore closed issue

Not really actionable since the reporter closed it because the project
“seems abandoned”. Good that cachix depends on it!

sternenseemann committed:
haskell.lib: make krank ignore informational issue references

sternenseemann committed:
haskell.lib.compose.buildStackProject: make krank ignore issue ref

Links the issue the line below addresses.

sternenseemann committed:
haskellPackages.cryptonite:let krank ignore issue reference in patch

The upstream issue reports closed state because the repository is
abandoned.

wolfgangwalther committed:
haskellPackages: fix transitive broken

wolfgangwalther committed:
haskellPackages.ihp-openai: unbreak (#445026)

sternenseemann committed:
haskellPackages.bloomfilter-blocked: disable test suite we can't easily build

(cherry picked from commit d19e1790d64df689daa077539e08496faf0034c9)

sternenseemann committed:
vaultenv: clarify that it won't be released to hackage

sternenseemann committed:
haskellPackages.spago: clarify that it won't be released to hackage

sternenseemann committed:
haskellPackages.cryptonite: make krank ignore closed issue tracker

cryptonite has been abandoned and all issues are reported as closed by
the GitHub API since the repository has been archived.

sternenseemann committed:
haskell.packages.ghc902Binary.reflection: make krank ignore closed issues

This workaround is for a GHC bug that has never gotten addressed in GHC 9.0.*.

sternenseemann committed:
haskell.packages.ghc902Binary.retry: drop obsolete override

The issues has been resolved in >= 0.9. Note that it's not possible to
test this at the moment since retry (indirectly) depends on
TemplateHaskell which doesn't work with our bindists.

sternenseemann committed:
haskell.packages.ghc902Binary.inline-c-cpp: remove broken/obsolete override

This depends on a GHC patch which the only remaining GHC 9.0.*,
ghc902Binary, doesn't have, being a bindist and all.

sternenseemann committed:
haskell.compiler.ghc902Binary: let krank ignore informational issue ref

sternenseemann commented:
@mpscholten Looks like a side effect of https://github.com/NixOS/nix/commit/88b7db1ba455926868dd61f5ea39e454e5fb0433, maybe it is worth discussing this with Nix upstream.

That being said, `unixPathMax 104` seems *wrong*, I wonder where it comes from. All I've seen from Googling is that `PATH_MAX` is typically 1024 on macOS which would be plenty.

sternenseemann committed:
haskell.packages.ghc910.{hinotify,fsnotify}: drop obsolete dontChecks

Upstream has since added proper 9.10 compatibility.

sternenseemann committed:
haskell.packages.ghc92.language-javascript_0_7_0_0: drop obsolete override

The issue with utf8-light appears to have been resolved and
haskell.packages./ghc92/.language-javascript_0_7_0_0 is completely unused.

sternenseemann commented:
The value is indeed correct: https://developer.apple.com/documentation/kernel/sockaddr_un/1541426-sun_path

sternenseemann commented:
@mpscholten Issue is migitated in Nix `master`: https://github.com/NixOS/nix/commit/7b22cd51052899b995f7427e587f0563fcda0637

nixpkgs-ci[bot] committed:
Merge 8e8a7943800de66a7e23052514dcb42ac4463546 into haskell-updates

sternenseemann committed:
spago-legacy: 0.21.0 -> 0.21.1, rename from spago

See https://github.com/purescript/spago-legacy/pull/3.

Co-authored-by: Peter Becich <peterbecich@gmail.com>
(cherry picked from commit 401566541bfa7ed6796cbcfbe7500e8ff5e70dce)

nixpkgs-ci[bot] committed:
Merge 77ac76b5ca173e9d8b9bd8798993798721f0b1e4 into haskell-updates

sternenseemann committed:
haskellPackages: stackage LTS 24.11 -> LTS 24.12

all-cabal-hashes: 2025-09-21T18:48:01Z -> 2025-09-25T14:09:07Z

(generated by maintainers/scripts/haskell/update-package-set.sh)

sternenseemann committed:
haskellPackages: postgresqlTestHook overrides to configuration-nix.nix

sternenseemann committed:
haskellPackages.postgresql-simple-migration: refactor override

sternenseemann committed:
haskellPackages: only use postgresqlTestHook if available

sternenseemann committed:
haskellPackages: disable remaining dependents on libsoup_2_4

sternenseemann committed:
ghcWithPackages: make krank ignore informational issue references

sternenseemann committed:
haskellPackages.turtle: fix mv not working on dirs

This fixes spago.tests, for example.

sternenseemann committed:
spago-legacy: move to by-name

(cherry picked from commit 2c29bb5c9db223645b86a96ada274f367979be0a)

sternenseemann commented:
Today is the last day of my rotation, tomorrow @maralorn will take over.

We've been able to merge the [update to 9.10](https://github.com/NixOS/nixpkgs/pull/429810) into staging-next in the last two weeks. This hasn't reached master yet and we should keep an eye on `staging-next` and look into extra regressions. Notable remaining regressions I know of:

- [ ] Stackage LTS updated the `gi-g*k*` packages to their Gtk 4 compatible versions. Some notable packages don't support this and haven't switched to the name spaced `gi-g*k3` packages yet. As a result e.g. `taffybar` (https://github.com/taffybar/taffybar/issues/606, https://github.com/NixOS/nixpkgs/issues/446879) and `termonad` (https://github.com/cdepillabout/termonad/issues/250, https://github.com/cdepillabout/termonad/issues/253, https://github.com/NixOS/nixpkgs/issues/446880) will be broken when `staging-next` is merged into `master`.
- [ ] A fix for `spago` is waiting for review at https://github.com/NixOS/nixpkgs/pull/324944

This PR is currently blocked on an issue introduced by #443159 which assumes that every Haskell package includes a cabal file at `${pname}.cabal`. This is not the case for e.g. `pkgs.naproche` and probably also violated downstream sometimes. See also [above](https://github.com/NixOS/nixpkgs/pull/445213#issuecomment-3340455459).

I want to fix this issue before/as @maralorn takes over and would like feedback on the approach to take. In principle, I think there are two possiblities:

1. In `postUnpack`, we search for `*.cabal` and store that for use in the generic builder. This should work, as Cabal requires that there is only a single `.cabal` file per directory.
2. We only run the `sed` removing CRLF when we are pulling in a Hackage revision. In this situation, we already assume that `${pname}.cabal` exists (which is somewhat reasonable). This is the main situation when we need the `sed` as well. However, it could hide issues with the `sed` since it would never run for unrevised package versions.

sternenseemann committed:
Merge commit e9f00bd893984bc8ce46c895c3bf7cac95 into haskell-updates

maralorn commented:
Why  not simply do
```
sed -i *.cabal -e 's/\r$//'
```

sternenseemann committed:
haskellPackages.mkDerivation: convert any/all cabal files to unix nls

In general, `pname` SHOULD be the same as the Cabal package name in
Haskell updates, but things work fine without this in many cases (e.g.
when only building executables), so we need to preserve this.

Since Cabal only allows a single .cabal file per directory, we can just
process *.cabal.

sternenseemann committed:
haskellPackages.mkDerivation: jailbreak any/all cabal files

Since Cabal only allows a single cabal file per directory, we can just
jailbreak *.cabal which works even if the user doesn't have set pname to
match the cabal file name (which they SHOULD do, though).

The only remaining reference to ${pname}.cabal in the revision handling
code is fine since we know the proper name of the cabal file from
Hackage in that case.

sternenseemann committed:
haskellPackages.mkDerivation: convert .cabal file after prePatch

cabal2nix emits prePatch = "hpack" for packages that need it, so the
.cabal file may not exist before prePatch.

sternenseemann committed:
spago: note why alias was introduced

(cherry picked from commit a3bce32ab065b3ca79f34c907f3a0d056cfd522c)

nixpkgs-ci[bot] committed:
Merge 62588b43e46ada6c53de392f4cd26dbc9c984026 into haskell-updates

mpscholten committed:
haskellPackages.ihp{,-ide,-migrate,-postgresql-simple-extra}: add maintainer mpscholten

maralorn committed:
haskellPackages.ihp{,-ide,-migrate,-postgresql-simple-extra}: add maintainer (#447190)

sternenseemann committed:
haskellPackages.microlens-pro: pin at 0.2.0.3

to match microlens < 0.5 prescribed by Stackage LTS 24.

sternenseemann committed:
haskellPackages.botan-bindings: drop released patch

sternenseemann committed:
fffuu: use sourceRoot instead of cd-ing in preCompileBuildDriver

(prePatch assumes it is executed in the same directory as the cabal
file.)

sternenseemann committed:
haskellPackages.vaultenv: run hpack in prePatch

This matches cabal2nix generated expressions. We assume that the cabal
file exists after the user supplied prePatch.

sternenseemann committed:
echidna: run hpack in prePatch

This matches cabal2nix generated expressions. We assume that the cabal
file exists after the user supplied prePatch.

sternenseemann committed:
haskellPackages.hexstring: restore Unix->DOS conversion of src file

This was removed in #443159 by mistake.

sternenseemann committed:
Merge commit 41b88f48c16c9f7821495ea8458d70c478 into haskell-updates

sternenseemann committed:
haskellPackages.zinza: update jailbreak comment

sternenseemann committed:
haskellPackages.granite: allow QuickCheck >= 2.15

sternenseemann committed:
haskellPackages.snappy-hs: work around erroneous vector constraint

sternenseemann committed:
haskellPackages.dataframe: drop obsolete override

This partially reverts commit d00493bd14a01107c279f251f8fcc264bc6a31a5.

sternenseemann committed:
dhall-lsp-server: drop now unnecessary dos2unix call

sternenseemann committed:
Merge commit 6e0e8f1225e6279dd301c20e3e069527e3 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 27ac93958969b5f3dccd654b402599cf3de633ac into haskell-updates

nixpkgs-ci[bot] committed:
Merge 0832d1b3a08fe4c695507a075d547ebfa818dfa0 into haskell-updates

wolfgangwalther committed:
Merge commit 'aee3f2ec9ef6ddea6115097946bc237a5b6ad8d9' into haskell-updates

sternenseemann commented:
The following PRs should fix most if not all remaining regressions in this PR:

- https://github.com/NixOS/nixpkgs/pull/447488
- https://github.com/NixOS/nixpkgs/pull/447025

sternenseemann committed:
pkgsCross.ghcjs.haskellPackages.reflex-dom: re-enable jobs

This is possible since the platform configuration is applied after
configuration-common.nix (which contains the dontDistributes).

wolfgangwalther committed:
haskell: fix missing cabal file after prePatch for various packages (#447488)

wolfgangwalther committed:
[haskell-updates] cherry picks from staging-next (#447025)

sternenseemann committed:
haskellPackages.haxl: work around broken Setup.hs

pkgs.codd depends on this package which wasn't previously apparent since
it is not part of our haskell-updates jobset.

sternenseemann committed:
haskell.packages.ghc9{4,6}.toml-reader: disable unsupported tests

skeletest only supports >= 9.6 for 0.1.0 and 9.4 not at all. Doesn't
seem worth it trying to get it to work for this.

sternenseemann committed:
Merge commit 59f9c6722b8f24b0014f222715cc315819 into haskell-updates

sternenseemann commented:
Changes from #444862 have been merged into this branch via 054e2d254162b5ee0f82299ff7557cb46a0aedcc. May be worth checking that no changes got lost in the merge.

nixpkgs-ci[bot] committed:
Merge 3ae8d45a094c060c47a44ce68afd90012d29a912 into haskell-updates

nixpkgs-ci[bot] committed:
Merge 53ba29feb0981cdb2d81c0b519dd9d95a226cdae into haskell-updates

raehik committed:
haskellPackages.HsOpenSSL-x509-system: fix Mac runtime

It's the same sort of platform-specific code as the
{crypton-,}x509-system packages. Applying the same bandaid fixes.

raehik committed:
haskellPackages: change darwin postPatch substitutes to use --replace-fail

nixpkgs-ci[bot] committed:
Merge fc430b0bdf9a58513b9f4607ef6f283f5c262e78 into haskell-updates

alexfmpe committed:
haskellPackages: fix build of cabal-macosx

nixpkgs-ci[bot] committed:
Merge 799d2d581d3d46d0dfe5b8f812a99580f1ce1095 into haskell-updates

rvl committed:
haskellPackages.gi-gtk-hs: use gi-gtk3 and gi-gdk3 to fix build

Upstream change that makes this change hasn't been released yet.

Co-authored-by: sternenseemann <sternenseemann@systemli.org>

nixpkgs-ci[bot] committed:
Merge 72229b769ba6b9551010e5b31d1ebad602dcd19b into haskell-updates

alexfmpe committed:
haskellPackages: fix executables built for ghcjs

maralorn committed:
haskellPackages: fix executables built for ghcjs (#451527)

wolfgangwalther committed:
[RFC] emanote: remove at 1.4.0.0 (#444083)

wolfgangwalther committed:
haskellPackages.HsOpenSSL-x509-system: fix Mac runtime (#450225)

alexfmpe committed:
haskellPackages: fix build of cabal-macosx (#450870)

wolfgangwalther committed:
haskellPackages: unbreak various packages

Created with unbreak.nu.

wolfgangwalther committed:
haskellPackages.haxl: work around broken Setup.hs (#449093)

nixpkgs-ci[bot] committed:
Merge cf2f6d331ea6734176d4f0183cbf234f204a9da9 into haskell-updates

wolfgangwalther committed:
haskellPackages.hledger_1_50_1: fix eval

Introduced via merge from master.

wolfgangwalther committed:
haskell-modules/HACKING: remove outdated reference to update hackage

Hackage is only updated directly from master, but not when merging the
PR anymore.

sternenseemann committed:
haskellPackages.gi-gtk-hs: Fix build (#451363)

wolfgangwalther committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1819351](https://hydra.nixos.org/eval/1819351) of nixpkgs commit [e1a3a9d](https://github.com/NixOS/nixpkgs/commits/e1a3a9d50a9412a716451506b981ed5c4544b035) as of 2025-10-15 09:00 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

sternenseemann commented:
> How would I do that?

Not so easy because GHC is nondeterministic. I've tried building HLS after a garbage collect which does seem to work for 9.6.

sternenseemann commented:
@alexfmpe https://hydra.nixos.org/build/310117246/nixlog/1

wolfgangwalther commented:
I have now marked packages as broken.

The following two packages fail to build, but I didn't mark them as broken, because we didn't ping maintainers in time:
- [ ] [[🐧❌]](https://hydra.nixos.org/build/310117246) [haskellPackages.proto3-suite](https://hydra.nixos.org/eval/1819351?filter=haskellPackages.proto3-suite) @alexfmpe
- [ ] [[🐧❌]](https://hydra.nixos.org/build/310117271) [tamarin-prover](https://hydra.nixos.org/eval/1819351?filter=tamarin-prover) @thoughtpolice

@alexfmpe @thoughtpolice you'll want to look into fixing your packages on `staging`.

For the two HLS builds for 9.6.7 and 9.8.4: I am hoping they will build fine on staging-next, when everything is rebuild anyway. If not, we need to look into that later.

<details>
<summary>Packages marked as broken</summary>

  - SyntaxMacros # failure in job https://hydra.nixos.org/build/309811610 at 2025-10-15
  - aes-gcm # failure in job https://hydra.nixos.org/build/309811668 at 2025-10-15
  - alpha # failure in job https://hydra.nixos.org/build/309811728 at 2025-10-15
  - amazonka-appconfigdata # failure in job https://hydra.nixos.org/build/309811755 at 2025-10-15
  - bindings-nettle # failure in job https://hydra.nixos.org/build/309812334 at 2025-10-15
  - calamity # failure in job https://hydra.nixos.org/build/309812724 at 2025-10-15
  - circuit-notation # failure in job https://hydra.nixos.org/build/309812708 at 2025-10-15
  - clash-lib # failure in job https://hydra.nixos.org/build/309812683 at 2025-10-15
  - clod # failure in job https://hydra.nixos.org/build/309812703 at 2025-10-15
  - convert-annotation # failure in job https://hydra.nixos.org/build/310117221 at 2025-10-15
  - dahdit # failure in job https://hydra.nixos.org/build/309813014 at 2025-10-15
  - data-foldapp # failure in job https://hydra.nixos.org/build/309813054 at 2025-10-15
  - data-list-zigzag # failure in job https://hydra.nixos.org/build/309813043 at 2025-10-15
  - derive-topdown # failure in job https://hydra.nixos.org/build/309813202 at 2025-10-15
  - engineering-units # failure in job https://hydra.nixos.org/build/309813391 at 2025-10-15
  - fastparser # failure in job https://hydra.nixos.org/build/309813509 at 2025-10-15
  - fxpak # failure in job https://hydra.nixos.org/build/309813706 at 2025-10-15
  - gmap # failure in job https://hydra.nixos.org/build/309814029 at 2025-10-15
  - haxl-amazonka # failure in job https://hydra.nixos.org/build/310117233 at 2025-10-15
  - haxl-facebook # failure in job https://hydra.nixos.org/build/310117231 at 2025-10-15
  - hbcd # failure in job https://hydra.nixos.org/build/309814355 at 2025-10-15
  - hgettext # failure in job https://hydra.nixos.org/build/309814473 at 2025-10-15
  - hint-nix # failure in job https://hydra.nixos.org/build/309814463 at 2025-10-15
  - ice40-prim # failure in job https://hydra.nixos.org/build/309814812 at 2025-10-15
  - json-rpc # failure in job https://hydra.nixos.org/build/309815009 at 2025-10-15
  - langchain-hs # failure in job https://hydra.nixos.org/build/309815178 at 2025-10-15
  - locators # failure in job https://hydra.nixos.org/build/309815296 at 2025-10-15
  - ltext # failure in job https://hydra.nixos.org/build/309815332 at 2025-10-15
  - monad-abort-fd # failure in job https://hydra.nixos.org/build/309815545 at 2025-10-15
  - murder # failure in job https://hydra.nixos.org/build/309815678 at 2025-10-15
  - nbparts # failure in job https://hydra.nixos.org/build/309815729 at 2025-10-15
  - ollama-holes-plugin # failure in job https://hydra.nixos.org/build/309815905 at 2025-10-15
  - reform-blaze # failure in job https://hydra.nixos.org/build/309816655 at 2025-10-15
  - robin-hood-profit # failure in job https://hydra.nixos.org/build/309816727 at 2025-10-15
  - roboservant # failure in job https://hydra.nixos.org/build/309816799 at 2025-10-15
  - sasha # failure in job https://hydra.nixos.org/build/309816856 at 2025-10-15
  - snappy # failure in job https://hydra.nixos.org/build/309817105 at 2025-10-15
  - sqlite-easy # failure in job https://hydra.nixos.org/build/309817187 at 2025-10-15
  - streamly-filepath # failure in job https://hydra.nixos.org/build/309817285 at 2025-10-15
  - telegram-bot-simple # failure in job https://hydra.nixos.org/build/309817498 at 2025-10-15
  - timeout-snooze # failure in job https://hydra.nixos.org/build/309817674 at 2025-10-15
  - typed-gui # failure in job https://hydra.nixos.org/build/309817848 at 2025-10-15
  - warp-tls-simple # failure in job https://hydra.nixos.org/build/309818122 at 2025-10-15
  - winio # failure in job https://hydra.nixos.org/build/309818237 at 2025-10-15
  - yabi # failure in job https://hydra.nixos.org/build/309818284 at 2025-10-15
</details>

wolfgangwalther committed:
Merge commit '886babdb9c62195fd4def9f30f8f690758aea3fd' into haskell-updates

sternenseemann commented:
We have multiple versions of ghc-exactprint in the dependency closure, though it is a mystery why it consistently works on our machines, but seemingly consistently doesn't on the Hydra builders.

```
Warning:
    This package indirectly depends on multiple versions of the same package. This is very likely to cause a compile failure.
      package haskell-language-server (haskell-language-server-2.11.0.0) requires ghc-exactprint-1.7.1.0-6x0Lt0OUX1n4DylHihsRdz
      package haskell-language-server (haskell-language-server-2.11.0.0) requires ghc-exactprint-1.7.1.0-6x0Lt0OUX1n4DylHihsRdz
      package haskell-language-server (haskell-language-server-2.11.0.0) requires ghc-exactprint-1.7.1.0-6x0Lt0OUX1n4DylHihsRdz
      package apply-refact (apply-refact-0.15.0.0-BLnycfa0wUu2RI1jkHhjpb) requires ghc-exactprint-1.7.1.0-6x0Lt0OUX1n4DylHihsRdz
      package retrie (retrie-1.2.3-EQ0AT5mQjO4ASN8zJrxz5Z) requires ghc-exactprint-1.7.1.0-kVpEFy1LJ04yZ61cXBTsb
```

sternenseemann commented:
tamarin-prover has been broken for a while and on master already, due to https://github.com/tamarin-prover/tamarin-prover/issues/774.

wolfgangwalther commented:
> We have multiple versions of ghc-exactprint in the dependency closure, though it is a mystery why it consistently works on our machines, but seemingly consistently doesn't on the Hydra builders.

Ah nice find. Should we try to fix this before merging?

wolfgangwalther committed:
haskell.packages.{ghc94,ghc96,ghc98}.haskell-language-server: disallow inconsistent dependencies

The diff is bigger than it needs to be, because I felt the urge to sort
things as well.

Key changes:
- For GHC 9.8, fix `super` -> `lself` for `extensions`.
- Add the `hls_overlay` to `apply-refact` as well.
- Remove `allowInconsistentDependencies`.

wolfgangwalther committed:
haskell.packages.{ghc94,ghc96,ghc98}.haskell-language-server: disallow inconsistent dependencies (#452213)
